Docs(Partner SDK): Truncate docs fix (#118)
fivetran/fivetran_partner_sdk: development-guide/destination-connector-development-guide.md

## FAQ

### Is it normal that, a source connector sends a truncate event followed by upsert event(s)?

> Yes, definitely. This is most likely an initial sync where the source connector first calls the `truncate` operation and then `upserts`. The `truncate` in this case, is meant to (soft) delete any rows that may have existed prior to the initial sync starting. This is done to make sure all rows that may have existed prior to the initial sync are marked as deleted (since we cannot be sure the initial sync will necessarily overwrite them all). It should pick out the rows that existed prior to the sync starting, in other words, where `_fivetran_synced` < "timestamp when `truncate` is called in the source connector".

Reviewer: ```suggestion
Yes, definitely. This happens during the [initial sync](https://fivetran.com/docs/getting-started/glossary#initialsync) or a [re-sync](https://fivetran.com/docs/using-fivetran/features#resync) where the source connector first calls the `truncate` operation and then `upserts`. The `truncate` in this case is meant to (soft) delete any rows that may have existed prior to the initial sync starting. This is done to make sure all rows that may have existed prior to the initial sync are marked as deleted (since we cannot be sure the initial sync will necessarily overwrite them all). It should pick out the rows that existed prior to the sync starting, in other words, where `_fivetran_synced` < "timestamp when `truncate` is called in the source connector".
```
